app/models/pieces/knight.rb

```diff
@@ -1,7 +1,33 @@
 class Knight < Piece
   before_create :set_defaults
-  
+
   def set_defaults
     self.icon = (self.color == 1) ? "&#9822;" : "&#9816;"
   end
-end
\ No newline at end of file
+
+  def valid_move?(x, y)
+    # Calculate old move and new move
+    x_move = (x_position - x).abs
+    y_move = (y_position - y).abs
+
+    # If x or y move is 2 then the next x or y move is 1
+    if x_move == 2 || y_move == 2
```

Comment:
Seems valid.. Can be refactored a little bit though:

you basically want to check if x_move, y_move == 2 and 1 or 1 and 2... you could add a method like:

def valid_move_combination?(x_move, y_move)
(x_move == 2 && y_move == 1) || (x_move == 1 && y_move ==2) ? true : false
end

then in valid_move? you define the x_move and y_move and call the new method..